main.go

```diff
@@ -129,7 +130,22 @@ func main() {
 		}
 		return nil
 	}
+	// If the command returns an error, cli takes upon itself to print
+	// the error on cli.ErrWriter and exit.
+	// Use our own writer here to ensure the log gets sent to the right location.
+	cli.ErrWriter = &FatalWriter{cli.ErrWriter}
 	if err := app.Run(os.Args); err != nil {
 		fatal(err)
 	}
 }
+
+type FatalWriter struct {
+	cliErrWriter io.Writer
+}
+
+func (f *FatalWriter) Write(p []byte) (n int, err error) {
+	msg := string(p)
+	logrus.Error(msg)
+	fmt.Fprint(f.cliErrWriter, msg)
+	return len(p), nil
```

Comment:
This should just be `f.cliErrWriter.Write(p)` IMO. Why go through `Fprintf` (which in languages like C would be punished with arbitrary code execution :wink:).